src/glossary.md

```diff
@@ -65,6 +73,13 @@ For example, `2 + (3 * 4)` is an expression that returns the value 14.
 An [item] that is not a member of an [implementation], such as a *free
 function* or a *free const*. Contrast to an [associated item].
 
+### Fundamental Type
+
+Includes `&`, `&mut`, `Box` and `Pin`. Any time a type `T` is
```

Comment:
Some points & questions (which @nikomatsakis may help with):

- `&` and friends are not types; they are type constructors.

- Suppose we have `#[fundamental] struct W<P0, P1> { ... }`. How would `W<A0, A1>` behave with respect to locality?  In this case the addition of a second type parameter is intentional so that we can suss out the *essence* of how fundamental types behave. It seems lifetimes have no effect on locality.

- Before itemizing the list of fundamental types it would be good to give a definition of how such types behave.

- What about fundamental `trait`s?